### DISCUSSION TITLE
Is dlib_face_recognition_resnet_model_v1 better than HSE_FaceRec_tf?

### DISCUSSION THREAD

**Discussion by kekoleg** [vladmandic/human#390] [General]
I am compairing [face-recognition.js](https://github.com/justadudewhohacks/face-recognition.js) (is using _dlib_face_recognition_resnet_model_v1_) and `human`. I have a set of 130 faces (different people) and another one that belongs to the person in this set and is used as a search `input`. Let's name the person we are looking for in the set `x`. I want to have `x` as the closest match to the `input`. Both `face-recognition.js` and `human` successfully found the searched person as the closest, but I've noticed one thing. The distance to the next person from the match is smaller in the `human` search result compared to the `face-recognition.js` search result. Does this mean that `face-recognition.js` was more confident in distinguishing faces and `human` will produce more false positives as the database grows significantly?

`human`
| Person | Distance | Match |
|--------|--------|--------|
| `x` | 7.644384539662718 | ✔️ |
|  | **10**.066655598143836 |  |
|  | 10.356641958367035 |  |
|  | 10.445426443843736 |  |
|  | 10.521632118242902 |  | 

`face-recognition.js`
| Person | Distance | Match |
|--------|--------|--------|
| `x` | 7.644384525550511 | ✔️ |
|  | **15**.685646888231423 |  |
|  | 16.320001040515428 |  |
|  | 16.522694225317679 |  |
|  | 17.079407024941559 |  | 

(sorted by smallest distance)

To compare the search results from both libraries I had to multiply every `face-recognition.js` distance by _27.3558887_. Am I doing something wrong? Am I jumping to the wrong conclusions?

**Comment by vladmandic** [comment]
sounds about right. but note that its a difficult topic.
- different face recognition models are trained on different face crop factors, so having wrong crop factor may show one model better than the other and changing crop factor may change it upside-down
- human supports several different face recognition models built-in

**Comment by kekoleg** [comment]
@vladmandic do you have any plans to support _dlib_face_recognition_resnet_model_v1_ in `human`? It's a really good high quality model, but `human` is generally much better than `face-recognition.js` in terms of resource consumption. Combining these things will be an excellent solution.

**Comment by vladmandic** [comment]
resnet models are generally much larger and not really suitable for web usage, so its not primary target for human.

**Reply by shurshilov** [reply]
How often does the person who created something not even understand how to use it. You are wrong, even if the model is 50 megabytes, it will be used, for example, in kiosk mode where the model is loaded once, but greater accuracy will improve everything

**Reply by vladmandic** [reply]
you're entitled to your opinion without the need to be insulting.

**Reply by shurshilov** [reply]
I didn't insult you, I just see very often when the creators of a library get into a dead end without understanding how to develop it, because don't know the application part. This is useful information for you and your product. But you're probably used to listening to only positive reviews... In general, by the way, false positives are also many. But this is another stage that has to be bypassed by checking the alignment of the eye position and so on. The recognition itself seems to be in error. Also anti-spoofing its just check blink eyes;)

**Reply by vladmandic** [reply]
You're assuming far too much about "usual developer" and never bothered to ask. That's fine, assume away. I'm not in habbit of pointless arguing, I'm out of this thread.
